server.go

```diff
@@ -112,7 +116,8 @@ func (s *Server) monitorHandler() {
 }
 
 func (s *Server) Handle(ctx context.Context, conn net.Conn) {
-	c := NewClient(conn, s.manager)
+	c := NewClient(ctx, conn, cfg)
+	c.consWg = &s.consumersWg
```

Comment:
I would prefer server and client having separate wait groups :) but we should address this in another iteration.

Reply:
Sure, let's revisit this in another PR!